supervisor/supervisor.go

```diff
@@ -48,9 +48,6 @@ func New(stateDir string, runtimeName, shimName string, runtimeArgs []string, ti
 	}
 	go s.exitHandler()
 	go s.oomHandler()
-	if err := s.restore(); err != nil {
```

Comment:
@mlaventure The point is to make sure health check works ASAP, so GRPC server has to be started ASAP, and GRPC services have be registered before server started, so we have to do restore after GRPC server starts since it usually takes a long time.

After registering the API we can accept GRPC requests, but these requests won't be handled before supervisor is started, and this PR guarantees that restore happens before supervisor starts. Do you think there'll be problem if the time between accepting GRPC requests and handling them is long, since we only have 2048 for tasks buffer?